### ISSUE TITLE
provisioning error when creating both a service bus namespace and event hub

### ISSUE THREAD

**Issue by bennage** [mspnp/data-pipeline#25]
When using the [Powershell script](/mspnp/data-pipeline/blob/master/src/Provisioning/ProvisionAssets.ps1) to provision assests, and when we are creating a new _Service Bus Namesapce_, I am consistently receiving the following:

``` 
C:\...\data-pipeline\src\Provisioning\CreateEventHub.ps1 : Exception calling
"CreateEventHubIfNotExists" with "1" argument(s): "The remote server returned an error: (401) Unauthorized.
InvalidSignature: The token has an invalid signature.
TrackingId:034dcd7d-9bf7-4a51-b914-b05aa4565002_G46,TimeStamp:1/18/2015 6:54:26 PM"
At C:\...\data-pipeline\src\Provisioning\ProvisionAssets.ps1:97 char:1
+ .\CreateEventHub.ps1 -Namespace $ServiceBusNamespace -Path $ServiceBusEventHubPa ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [CreateEventHub.ps1], MethodInvocationException
    + FullyQualifiedErrorId : UnauthorizedAccessException,CreateEventHub.ps1
````

If the namespace already exists, then the script executes as expected.

**Comment by mj1856** [comment]
I tried this a few times (starting clean each time).  Interestingly, it worked once, failed once with he same message you got, and then failed again with a different error:

```
D:\dev\pnp\data-pipeline\src\Provisioning\CreateEventHub.ps1 : Exception calling "CreateEventHubIfNotExists" with "1"
argument(s): "The remote server returned an error: (500) Internal Server Error. SubCode=50000. Internal  Server Error.
TrackingId:39bb2539-72bb-4a88-870e-08f95658c541_G23,TimeStamp:1/20/2015 5:33:45 PM"
At D:\dev\pnp\data-pipeline\src\Provisioning\ProvisionAssets.ps1:97 char:1
+ .\CreateEventHub.ps1 -Namespace $ServiceBusNamespace -Path $ServiceBusEventHubPa ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [CreateEventHub.ps1], MethodInvocationException
    + FullyQualifiedErrorId : MessagingException,CreateEventHub.ps1
```

Perhaps there's a time delay needed between creation of the service bus and creation of the event hub?  I'll play with it more later and let you know my findings.

**Comment by bennage** [comment]
Thanks!

**Comment by mj1856** [comment]
I was going to work on this tonight, but I can no longer reproduce the error.  Perhaps a service glitch?  Not sure.  Can you still get it to happen?

**Comment by bennage** [comment]
We'll reopen if the problem is reported again.

**Comment by drcrook1** [comment]
Im having same issue.

**Comment by fsimonazzi** [comment]
According to the docs (https://msdn.microsoft.com/en-us/library/azure/hh690931.aspx) it takes time for the namespace's resources to be provisioned. We've added a "wait until async" step in our provisioning code.

**Comment by fsimonazzi** [comment]
I've tried adding a "wait until active" step to the provisioning script but still got the 401 error

**Comment by ammaroff** [comment]
It always fails with me if you I haven't Service Bus already.
Second time  always success

**Comment by bennage** [comment]
Yes, we've had a hard time making this reliable. Unfortunately, we're stuck at the moment.
________________________________

**Comment by vicky23** [comment]
Hi

Exception calling "CreateEventHubIfNotExists" with "1" argument(s): "The remote server returned an error: (401) Unauthorized. 
InvalidSignature: The token has an invalid signature. TrackingId:f13798d9-b3a2-4728-94b3-85fb5e481f24_G18,TimeStamp:12/18/2015 
10:43:05 AM"

I am getting this error continuously for past one hour it worked fine earlier but not now. Is it related to server down or any other problem

**Comment by SoulOfUniverse** [comment]
Its bloody Azure changes the game, you have to generate new shared access policy and include Manage policy, their connection string manager generates wrong connection string, for some reason it generates the default one for your topic name, it is still buggy as hell, it used to generate correct conncetion string of shared access policy for whole ServiceBus not for Topic individually.
